Src/RE2.Managed/Regex2.cs

```diff
@@ -186,6 +187,90 @@ public static void ClearRegexes()
             }
         }
 
+        public static unsafe void GetRegexSetup(string pattern, out ulong numCapturingGroups, out ulong numNamedCapturingGroups, out ulong numSubmatches, out ulong groupNameHeadersBufferSize, out ulong groupNamesBufferSize, out ulong submatchesBufferSize)
+        {
+            byte[] patternUtf8Bytes = Encoding.UTF8.GetBytes(pattern);
+
+            fixed (byte* patternUtf8BytesPtr = patternUtf8Bytes)
+            {
+                fixed (ulong* numCapturingGroupsPtr = &numCapturingGroups)
+                fixed (ulong* numNamedCapturingGroupsPtr = &numNamedCapturingGroups)
+                fixed (ulong* numSubmatchesPtr = &numSubmatches)
+                fixed (ulong* groupNameHeadersBufferSizePtr = &groupNameHeadersBufferSize)
+                fixed (ulong* groupNamesBufferSizePtr = &groupNamesBufferSize)
+                fixed (ulong* submatchesBufferSizePtr = &submatchesBufferSize)
+                {
+                    NativeMethods.GetRegexSetup(new StringUtf8(patternUtf8BytesPtr, (ulong)pattern.Length), numCapturingGroupsPtr, numNamedCapturingGroupsPtr, numSubmatchesPtr, groupNameHeadersBufferSizePtr, groupNamesBufferSizePtr, submatchesBufferSizePtr);
+                }
+            }
+        }
+
+        public static unsafe bool Matches4(
```

Comment:
>Matches4 [](start = 34, length = 8)

lets use Matches. Since the parameters are different, the overload would work.

Reply:
Yes, the name and parameters will probably change after refactoring.

Reply:
For now, I've changed it to a `Matches` overload. This API may change after refactoring though.

Reply:
General refactoring moved to backlog: [https://github.com/microsoft/sarif-pattern-matcher/issues/386](https://github.com/microsoft/sarif-pattern-matcher/issues/386)